src/pandapipes/pf/result_extraction.py

```diff
@@ -212,6 +213,11 @@ def extract_branch_results_with_internals(net, branch_results, table_name,
 
             for i, (res_name, entry) in enumerate(res_mean_hydraulics):
                 res_table[res_name].values[pt] = res[i + 3][connected_ind] / num_internals
+    #outlet temperature
+    use_numba = get_net_option(net, "use_numba")
+    indices, sections = _sum_by_group(use_numba, branch_pit[:, ELEMENT_IDX], np.ones(len(branch_pit)))
+    indices_last_section = (np.cumsum(sections)-1).astype(int)
+    res_table["t_outlet_k"] = branch_results["t_outlet"][indices_last_section]
```

Comment:
I can imagine that there are still some tests failing that check indexing. Please have another look at how this is done in other sections. You need to consider the connected_thermal lookup and you should work on the underlying numpy values of the result table and need to match the pit indices to the result table indices.